clang/lib/AST/OpenACCClause.cpp

```diff
@@ -166,3 +176,9 @@ void OpenACCClausePrinter::VisitVectorLengthClause(
     const OpenACCVectorLengthClause &C) {
   OS << "vector_length(" << C.getIntExpr() << ")";
 }
+
+void OpenACCClausePrinter::VisitPrivateClause(const OpenACCPrivateClause &C) {
+  OS << "private(";
+  llvm::interleaveComma(C.getVarList(), OS);
```

Comment:
Do you have a test for this?

Reply:
I dont!  I THINK this uses 'ast-print', so give me a minute and I'll put one on this review that tests all of the variants here.

Reply:
Test added :)  A bit of exploration, plus a patch to fix the rest (upstreamed as review after commit), and now up to date here :)